nixos/lib/test-driver/test-driver.py

```diff
@@ -827,13 +934,14 @@ def process_serial_output() -> None:
     def cleanup_statedir(self) -> None:
         if os.path.isdir(self.state_dir):
             shutil.rmtree(self.state_dir)
```

Comment:
rmtree checks if the path is a directory, and if the path is a file, we might as well throw an exception now rather than waiting for something else to fail (eg `mkdir(path)` in `__init__`)
```suggestion
        shutil.rmtree(self.state_dir)
```